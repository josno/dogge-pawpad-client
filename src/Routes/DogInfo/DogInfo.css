.dog-info {
	height: 100%;
}

.dog-name {
	display: flex;
	flex-direction: column;
}

.dog-name-text {
	color: #f7567c;
	font-family: Candal, sans-serif;
	font-size: 3em;
	margin: 5px;
	-webkit-text-stroke: 1px black;
}

.dog-info-text {
	line-height: 25px;
	margin: 20px;
	text-align: left;
}

.shot-container {
	display: flex;
	flex-direction: column;
	flex-wrap: wrap;
	height: 100%;
}

.grid-container {
	display: flex;
	flex-direction: column;
	flex-wrap: wrap;
	width: 100%;
}

.indicator-no {
	color: #872f44;
}

.indicator-yes {
	color: #306b34;
}

.info-img {
	border-radius: 50%;
	width: 250px;
	height: 250px;
	object-fit: cover;
	border: 2px solid black;
}

.info-title {
	color: #5d576b;
	margin: 5px;
}

.last-shot-text {
	display: block;
	padding-left: 30px;
	font-size: 0.75em;
	margin: 0px;
}

.box-flex,
.shot-check-container {
	display: flex;
	flex-direction: column;
}

.dog-img {
	margin: 20px;
}

.box-flex {
	background-color: #fcfcfc;
	margin: 20px;
	border: 2px solid;
}

.details-grid-container {
	background-clip: content-box;
	display: grid;
	grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr;
	grid-template-columns: 0.9fr 1fr;
	grid-template-areas: "gender gender-value" "age age-value" "arrival arrival-value" "tag tag-value" "microchip microchip-value" "spayed-neutered spayed-neutered";
	height: 100%;
}

.align-details {
	align-content: flex-start;
	align-items: center;
	display: flex;
	font-size: 0.7em;
}

.gender {
	grid-area: gender;
}

.gender-value {
	grid-area: gender-value;
}

.age {
	grid-area: age;
}

.age-value {
	grid-area: age-value;
}

.arrival {
	grid-area: arrival;
}

.arrival-value {
	grid-area: arrival-value;
}

.tag {
	grid-area: tag;
}

.tag-value {
	grid-area: tag-value;
}

.microchip {
	grid-area: microchip;
}

.microchip-value {
	grid-area: microchip-value;
}

.spayed-neutered {
	grid-area: spayed-neutered;
}

@media (min-width: 800px) {
	.align-details {
		font-size: 1em;
	}

	.grid-container {
		background-clip: content-box;
		padding: 50px;
		display: grid;
		grid-template-columns: 1fr 1.1fr;
		grid-template-rows: 0.4fr 0.1fr;
		grid-template-areas: "dog-name nav-buttons" "basic-dog-details basic-dog-details" "shots-information shots-information";
	}

	.basic-dog-details {
		grid-area: basic-dog-details;
	}

	.dog-img {
		grid-area: dog-img;
		height: 300px;
	}

	.shots-information {
		grid-area: shots-information;
	}

	.shot-check-container {
		display: flex;
		justify-content: space-evenly;
	}
}
@media (min-width: 1200px) {
	.grid-container {
		max-width: 700px;
		margin-left: auto;
		margin-right: auto;
	}
	/* .shot-container {
		height: 200px;
	} */
	.dog-info-text {
		font-size: 0.8em;
	}
}
