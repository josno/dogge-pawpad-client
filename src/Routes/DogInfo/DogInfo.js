import React, { Component } from "react";
import { Modal } from "react-responsive-modal";
import "react-responsive-modal/styles.css";
import { Link } from "react-router-dom";
import FosterAdopForm from "../../Components/FosterAdopForm/FosterAdopForm";
import ArchiveModal from "../../Components/ArchiveModal/ArchiveModal";
import ModalForm from "../../Components/ImgModalForm/ImgModalForm";

import DogDetailsView from "../../Components/DogDetailsView/DogDetailsView";
import ShotDetailsView from "../../Components/ShotDetailsView/ShotDetailsView";
import PawPadContext from "../../PawPadContext.js";
import DogsApiService from "../../services/api-service";
import AdoptionDetails from "../../Components/AdoptionDetails/AdoptionDetails";
import EditShots from "../../Components/EditShots/EditShots";

import "./DogInfo.css";
import moment from "moment";

class DogInfo extends Component {
	static contextType = PawPadContext;
	constructor(props) {
		super(props);
		this.state = {
			dogInfo: "",
			openAdopt: false,
			openFoster: false,
			openArchive: false,
			openProfileImg: false,
			editShotMode: false,
			error: null,
		};
		this.formatDate = this.formatDate.bind(this);
		this.handleDelete = this.handleDelete.bind(this);
		this.handleArchive = this.handleArchive.bind(this);
		this.updateDogImage = this.updateDogImage.bind(this);
		this.handleDogAdoption = this.handleDogAdoption.bind(this);
		this.handleDogFoster = this.handleDogFoster.bind(this);
		this.changeShotEditMode = this.changeShotEditMode.bind(this);
	}

	formatDate(date) {
		const formattedDate = moment(date).format("LL");
		return formattedDate;
	}

	openModal = (e) => {
		const { name } = e.target;
		this.setState({ [name]: true });
	};

	closeModal = (str) => {
		this.setState({ [str]: false });
	};

	handleDelete = () => {
		const { dogId } = this.props.match.params;

		DogsApiService.deleteDog(dogId).then((response) => {
			DogsApiService.deleteNotesByDogId(this.props.match.params.dogId);
			DogsApiService.deleteShotsByDogId(this.props.match.params.dogId);
			this.props.history.push("/dogs-list");
		});
	};

	changeShotEditMode = () => {
		const mode = this.state.editShotMode;
		this.setState({
			editShotMode: !mode,
		});
	};

	handleArchive = (str) => {
		const { dogId } = this.props.match.params;
		const dateObj = { archive_date: new Date() };
		const noteObj = {
			type_of_note: "archive",
			notes: str,
			dog_id: dogId,
		};
		DogsApiService.archiveDog(dogId, dateObj)
			.then((response) => DogsApiService.insertNewNote(noteObj))
			.then((response) => {
				DogsApiService.getDogInfo(dogId).then((res) =>
					this.setState({
						openArchive: false,
						dogInfo: res,
					})
				);
			})
			.catch((err) => this.setState({ error: "Can't archive dog." }));
	};

	handleDogAdoption = () => {
		const { dogInfo } = this.state;

		this.setState({
			dogInfo: { ...dogInfo, dog_status: "Adopted" },
			openAdopt: false,
		});
	};

	handleDogFoster = () => {
		const { dogInfo } = this.state;
		this.setState({
			dogInfo: { ...dogInfo, dog_status: "Fostered" },
			openFoster: false,
		});
	};

	updateDogImage(e, profileImg) {
		e.preventDefault(e);
		const profile_img = profileImg;

		const formData = new FormData();
		formData.append("profile_img", profile_img);

		DogsApiService.deleteDogImg(formData, this.state.dogInfo.tag_number)
			.then((res) => {
				return DogsApiService.uploadDogImg(
					formData,
					this.state.dogInfo.tag_number
				);
			})
			.then((res) => {
				const dogObj = {
					dog_name: this.state.dogInfo.dog_name,
					profile_img: res,
				};
				return DogsApiService.updateDog(dogObj, this.state.dogInfo.id);
			})
			.then((res) => {
				return DogsApiService.getDogInfo(this.state.dogInfo.id);
			})
			.then((res) => {
				this.setState({
					dogInfo: res,
					openProfileImg: false,
				});
			})
			.catch((err) => this.setState({ error: err }));
	}

	renderNavButtons = (dogInfo) => {
		return (
			<div className="nav-buttons">
				{dogInfo.dog_status === "Adopted" && (
					<Link
						className="delete"
						to={{
							pathname: `/adoption-details-${dogInfo.dog_name}/${dogInfo.id}`,
							state: dogInfo,
						}}
					>
						Adoption Details
					</Link>
				)}
				{dogInfo.dog_status === "Current" && (
					<>
						<button
							className="delete"
							name="openAdopt"
							onClick={(e) => this.openModal(e)}
						>
							Adopted
						</button>

						<button
							className="delete"
							name="openFoster"
							onClick={(e) => this.openModal(e)}
						>
							Fostered
						</button>
					</>
				)}

				<button className="see-notes">
					<Link
						className="dog-link"
						to={`/notes-${dogInfo.dog_name}/${dogInfo.id}`}
					>
						Notes
					</Link>
				</button>

				{dogInfo.dog_status !== "Archived" && (
					<button
						className="delete"
						name="openArchive"
						onClick={(e) => this.openModal(e)}
					>
						Archive
					</button>
				)}

				{dogInfo.dog_status !== "Adopted" && (
					<button className="delete" onClick={this.handleDelete}>
						Delete
					</button>
				)}
			</div>
		);
	};

	renderModals = (dogInfo) => {
		return (
			<>
				<Modal
					open={this.state.openProfileImg}
					onClose={(e) => this.closeModal("openProfileImg")}
					center
				>
					<ModalForm handleUpdate={this.updateDogImage} />
				</Modal>
				<Modal
					open={this.state.openArchive}
					onClose={(e) => this.closeModal("openArchive")}
					center
				>
					<ArchiveModal
						dogName={dogInfo.dog_name}
						dogId={dogInfo.dogId}
						handleArchive={this.handleArchive}
					/>
				</Modal>
				<Modal
					open={this.state.openAdopt}
					onClose={(e) => this.closeModal("openAdopt")}
					center
				>
					<FosterAdopForm
						type="adopt"
						dogId={dogInfo.id}
						updateDogInfo={this.handleDogAdoption}
					/>
				</Modal>
				<Modal
					open={this.state.openFoster}
					onClose={(e) => this.closeModal("openFoster")}
					center
				>
					<FosterAdopForm
						type="foster"
						dogId={dogInfo.id}
						updateDogInfo={this.handleDogFoster}
					/>
				</Modal>
			</>
		);
	};

	renderAdoptionDetails = (id) => {
		return (
			<div className="adoption-details box-flex">
				<AdoptionDetails dogId={id} />
			</div>
		);
	};

	renderDogImgName = (obj) => {
		return (
			<div className="dog-name">
				<img alt="dog-name" className="info-img" src={obj.profile_img} />
				<button
					className="edit-pencil edit-pencil-img"
					name="openProfileImg"
					onClick={(e) => this.openModal(e)}
				>
					&#9998;
				</button>
				<h1 className="dog-name-text">{obj.dog_name}</h1>
			</div>
		);
	};

	renderDogDetailsView = (props) => {
		return <DogDetailsView dogId={props.dogId} />;
	};

	renderShots = (shots) => {
		return (
			<div className="shots-information box-flex">
				<h3 className="info-title">Shots Completed</h3>
				<button
					className="edit-shot-pencil edit-pencil"
					onClick={(e) => this.changeShotEditMode(e)}
				>
					&#9998;
				</button>
				{!this.state.editShotMode ? (
					<>
						<ul className="dog-info-text shot-container">
							<ShotDetailsView shots={shots} />
						</ul>
					</>
				) : (
					<EditShots dogId={this.props.match.params.dogId} />
				)}
			</div>
		);
	};

	renderUpdateByLine = (obj) => {
		return !obj.updated_by ? (
			""
		) : (
			<div className="updated-by">
				<p>
					Updated by {obj.updated_by} on{" "}
					{this.formatDate(obj.notes_date_modified)}
				</p>
			</div>
		);
	};

	async componentDidMount() {
		const { dogId } = this.props;
		const res = await DogsApiService.getDogInfo(dogId);

		this.setState({
			dogInfo: res,
			shots: res.shotsCompleted,
		});
	}

	render() {
		const { dogInfo, shots } = this.state;

		return (
			<main className="dog-info">
				<div className="grid-container">
					{this.renderDogImgName(dogInfo)}
					{this.renderNavButtons(dogInfo)}
					{this.renderModals(dogInfo)}
					{this.renderDogDetailsView(this.props)}
					{this.renderShots(shots)}
				</div>
				{this.renderUpdateByLine(dogInfo)}
			</main>
		);
	}
}

export default DogInfo;
