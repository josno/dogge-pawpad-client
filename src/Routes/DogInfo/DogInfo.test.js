import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import DogInfo from './DogInfo';

describe('<DogInfo />', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(
			<BrowserRouter>
				<DogInfo />
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<DogInfo />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
