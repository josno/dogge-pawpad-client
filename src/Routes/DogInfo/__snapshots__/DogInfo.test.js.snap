// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DogInfo /> renders as expected 1`] = `
<main
  className="dog-info"
>
  <div
    className="grid-container"
  >
    <div
      className="dog-name"
    >
      <img
        alt="dog-name"
        className="info-img"
      />
      <button
        className="edit-pencil edit-pencil-img"
        name="openProfileImg"
        onClick={[Function]}
      >
        ✎
      </button>
      <h1
        className="dog-name-text"
      />
    </div>
    <div
      className="nav-buttons"
    >
      <button
        className="see-notes"
      >
        <a
          className="dog-link"
          href="/notes-undefined/undefined"
          onClick={[Function]}
        >
          Notes
        </a>
      </button>
      <button
        className="delete"
        name="openArchive"
        onClick={[Function]}
      >
        Archive
      </button>
      <button
        className="delete"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
    <div
      className="basic-dog-details box-flex"
    >
      <h3
        className="info-title"
      >
        Basic Details 
      </h3>
      <ul
        className="dog-info-text details-grid-container"
      >
        <li
          className="gender align-details"
        >
          Gender:
          <div
            className="gender-value align-details"
          >
            <label
              htmlFor="male"
            >
              <input
                id="male"
                name="gender"
                onChange={[Function]}
                required={true}
                type="radio"
                value="Male"
              />
              Male
            </label>
            <label
              htmlFor="female"
            >
              <input
                id="female"
                name="gender"
                onChange={[Function]}
                type="radio"
                value="Female"
              />
              Female
            </label>
            <button
              onClick={[Function]}
              value="gender"
            >
              ✘
               
            </button>
            <button
              onClick={[Function]}
              value="gender"
            >
              ✔
               
            </button>
          </div>
        </li>
        <li
          className="age align-details"
        >
          Birthdate:
          <div
            className="age-value align-details"
          >
            Invalid date
            <button
              className="edit-pencil"
              onClick={[Function]}
              value="age"
            >
              ✎
            </button>
          </div>
        </li>
        <li
          className="arrival_date align-details"
        >
          Arrival Date:
          <div
            className="arrival_date-value align-details"
          >
            Invalid date
            <button
              className="edit-pencil"
              onClick={[Function]}
              value="arrival_date"
            >
              ✎
            </button>
          </div>
        </li>
        <li
          className="tag_number align-details"
        >
          Tag:
          <div
            className="tag_number-value align-details"
          >
            
            <button
              className="edit-pencil"
              onClick={[Function]}
              value="tag_number"
            >
              ✎
            </button>
          </div>
        </li>
        <li
          className="microchip align-details"
        >
          Microchip:
          <div
            className="microchip-value align-details"
          >
            
            <button
              className="edit-pencil"
              onClick={[Function]}
              value="microchip"
            >
              ✎
            </button>
          </div>
        </li>
        <li
          className="spayedneutered align-details"
        >
          Spayed/Neutered:
          <div
            className="spayedneutered-value align-details"
          >
             
            <span
              className="indicator-no"
            >
               ✘ 
            </span>
            <button
              className="edit-pencil"
              onClick={[Function]}
              value="spayedneutered"
            >
              ✎
            </button>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="shots-information box-flex"
    >
      <h3
        className="info-title"
      >
        Shots Completed
      </h3>
      <button
        className="edit-shot-pencil edit-pencil"
        onClick={[Function]}
      >
        ✎
      </button>
      <ul
        className="dog-info-text shot-container"
      >
        <div>
          <ul
            className="dog-info-text shot-container"
          />
        </div>
      </ul>
    </div>
  </div>
  
</main>
`;
